tests/parametric/test_config_consistency.py

```diff
@@ -43,3 +43,34 @@ def test_tracing_disabled(self, library_env, test_agent, test_library):
         assert (
             True
         ), "wait_for_num_traces raises an exception after waiting for 1 trace."  # wait_for_num_traces will throw an error if not received within 2 sec, so we expect to see an exception
+
+
+def set_service_version_tags():
+    env1 = {}
+    env2 = {"DD_SERVICE": "test_service", "DD_VERSION": "5.2.0"}
+    return parametrize("library_env", [env1, env2])
+
+
+@scenarios.parametric
+@features.tracing_configuration_consistency
+class Test_Config_UnifiedServiceTagging:
+    @set_service_version_tags()
+    def test_version_tag(self, library_env, test_agent, test_library):
+        assert library_env.get("DD_SERVICE", "test_service") == "test_service"
+        assert library_env.get("DD_VERSION", "5.2.0") == "5.2.0"
```

Comment:
This line looks like it will fail for env1 because it has an empty set of environment variables. Have you tested this locally?

Reply:
It doesn't fail because it the `.get(key, val)` defaults the return value to the `val` if it is not found in `library_env`.

Reply:
So you're saying lines 59 & 60 will pass even with `env1`? That feels **wrong** 😆

Reply:
Hmm, this is done elsewhere as well, such as `tests/parametric/test_dynamic_configuration.py::TestDynamicConfigTracingEnabled::test_tracing_client_tracing_enabled`. 

I could turn it into an if statement to check both conditions and assert them. Do you think that is a more sound way to implement this check?

Reply:
Hmm.. maybe you don't even need these assertions -- like they're just checking that the env vars (that you're setting above) were set. But we can assume they were set because you set them 😆 .

The issue with keeping it is it makes the expected behavior when DD_SERVICE and DD_VERSION are not set ambiguous -- is it that val was just returned because of the get() function's default return, or was val returned because we really determined that's what the env var is set to?

Ultimately, this test may not be clear enough with the distinct behaviors we expect. The `env1` condition could theoretically pass on line 73/74, even though we'd want it to pass on 75/76 instead. You've made it all nice and succinct, but maybe the test will be clearer and more robust if you create separate functions for each condition.

Reply:
I split the two cases into separate functions and directly check their conditions in the new commit.